lib/featureDetection.js

```diff
@@ -27,17 +30,38 @@ function getConfigVarFlag(stdout) {
   return '-Des.';
 }
 
-exports.getFeatures = memoize(function featureDetect(path) {
-  var cmd = join(path, 'bin', 'elasticsearch');
 
+function getFeaturesFromBin(path) {
+  var cmd = join(path, 'bin', 'elasticsearch');
   return exec(cmd + ' -h')
   .then(function (results) {
     var stdout = results[0];
     var stderr = results[1];
 
     return {
       versionFlag: getVersionFlag(stdout, stderr),
-      configVarFlag: getConfigVarFlag(stdout, stderr)
+      configVarFlag: getConfigVarFlag(stdout, stderr),
+
     };
   });
+}
+function getFeaturesFromFile(path) {
+  return access(join(path, 'config', 'log4j2.properties'), F_OK)
+  .then(function () {
+    return {
+      log4j2Flag: true
```

Comment:
The other features are labeled with `Flag` because they detect the command line flags the elasticsearch bin accepts. This should probably just be `usesLoggingYaml` or something.